### PR TITLE
button columns support added in generic button v3

### PR DESCRIPTION
**Pull request by shivanshu877**
### Information

- Jira: YT-<Number>
- Link of PR pointing to `@develop`: 
  
### I've done the following things before and after raising the PR

* Put the JIRA number `[YT-NNNN]` in the PR title.
* Done a self review of the PR.
* Assigned reviewers to this PR.

### CODE REVIEW

File: lib/ui/components/generic_button/generic_button_v3.widget.dart [hunk 1]
```diff
@@ -2,6 +2,106 @@ import 'package:flutter/material.dart';
...
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: List.generate(
+        buttonDetails.buttonRows!.length,
+        (rowIndex) {
```
Comment (Suryanshomar7240) [inline lib/ui/components/generic_button/generic_button_v3.widget.dart:19]: move in different file

File: lib/ui/components/generic_button/generic_button_v3.widget.dart [hunk 2]
```diff
@@ -2,6 +2,106 @@ import 'package:flutter/material.dart';
...
+    required this.buttonDetails,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: List.generate(
+        buttonDetails.buttonRows!.length,
```
Comment (Suryanshomar7240) [inline lib/ui/components/generic_button/generic_button_v3.widget.dart:18]: handle null check

Comment (Suryanshomar7240) [reply lib/ui/components/generic_button/generic_button_v3.widget.dart:18]: `buttonDetails.buttonRows?.length??0`

File: lib/ui/components/generic_button/generic_button_v3.widget.dart [hunk 3]
```diff
@@ -2,6 +2,106 @@ import 'package:flutter/material.dart';
...
+        (rowIndex) {
+          final buttonRow = buttonDetails.buttonRows![rowIndex];
+
+          if (buttonRow.buttonColumns != null) {
+            return Row(
+              mainAxisAlignment: buttonRow.flexAlignment,
+              children: List.generate(
+                buttonRow.buttonColumns!.length,
+                (colIndex) => StyledComponentWidget(
+                  styledComponent: buttonRow.buttonColumns![colIndex],
```
Comment (Suryanshomar7240) [inline lib/ui/components/generic_button/generic_button_v3.widget.dart:28]: handle null check

**Review by Suryanshomar7240 — CHANGES_REQUESTED** [review]

File: lib/ui/components/generic_button/generic_button_v3.widget.dart (hunk 1 above)
Comment (shivanshu877) [reply lib/ui/components/generic_button/generic_button_v3.widget.dart]: done

File: lib/ui/components/generic_button/generic_button_v3.widget.dart (hunk 2 above)
Comment (shivanshu877) [reply lib/ui/components/generic_button/generic_button_v3.widget.dart]: handled

File: lib/ui/components/generic_button/generic_button_v3.widget.dart (hunk 3 above)
Comment (shivanshu877) [reply lib/ui/components/generic_button/generic_button_v3.widget.dart]: done

File: lib/ui/components/generic_button/generic_button_v3.widget.dart [hunk 4]
```diff
@@ -80,87 +80,8 @@ class GenericButtonV3Widget extends StatelessWidget {
...
-                                          ),
-                                        ),
-                                      )
-                                      .toList(),
-                                )
-                              : const SizedBox.shrink(),
-                        )
-                        .toList(),
+                ? GetGenericButtonV3Rows(
+                    buttonDetails: buttonDetails,
```
Comment (Suryanshomar7240) [inline lib/ui/components/generic_button/generic_button_v3.widget.dart:84]: GenericButtonV3RowsWidget
`GenericButtonV3RowsWidget
`

**Review by Suryanshomar7240 — CHANGES_REQUESTED** [review]

File: lib/ui/components/generic_button/generic_button_v3.widget.dart (hunk 4 above)
Comment (shivanshu877) [reply lib/ui/components/generic_button/generic_button_v3.widget.dart]: done

**Review by Suryanshomar7240 — APPROVED** [review]

**Pull request merged**